base/windows/shop/BuyArtifactsMenu.js

```diff
@@ -0,0 +1,437 @@
+                this.npc_dialog.update_dialog("buy_quantity");
+                let shop_items = this.data.info.shops_list[this.parent.shop_key].item_list;
+                let shop_item_match = shop_items.filter(i => { return (i.key_name === this.selected_item.key_name); })[0];
+                let shop_item = {key_name: shop_item_match.key_name, quantity: shop_item_match.quantity === -1 ? 30 : shop_item_match.quantity};
+
+                let char_item_match = this.selected_character.items.filter(i => { return (i.key_name === this.selected_item.key_name); });
+                let char_item = char_item_match.length !== 0 ? char_item_match[0] : null;
+
+                if(!this.quant_win.is_open) this.quant_win.open(shop_item, char_item);
+                this.control_manager.set_control(true, false, true, false, {right: this.quant_win.increase_amount.bind(this.quant_win),
+                    left: this.quant_win.decrease_amount.bind(this.quant_win),
+                    esc: this.open_inventory_view.bind(this),
+                    enter: this.on_purchase_success.bind(this)});
+            }
+        }
+    }
+
+    on_cancel_char_select(){
+        if(this.inv_win.is_open) this.inv_win.close();
+        if(this.eq_compare.is_open) this.eq_compare.close();
+        if(this.char_display.is_open)this.char_display.close();
+        this.open_buy_select();
+    }
+
+    on_cancel_game_ticket(){
+        this.npc_dialog.update_dialog("game_ticket_decline", true);
+        this.control_manager.set_control(false, false, false, false, {esc: this.on_cancel_char_select.bind(this),
+            enter: this.on_cancel_char_select.bind(this)});
+    }
+
+    open_equip_compare(){
+        this.buy_select_pos = {page: this.buy_select.current_page, index: this.buy_select.selected_index};
+        if(this.item_desc_win.open) this.item_desc_win.close();
+        if(this.buy_select.is_open) this.buy_select.close();
+        this.npc_dialog.update_dialog("character_select");
+
+        let char_key = (this.selected_character) ? this.selected_character.key_name : this.data.info.party_data.members[0].key_name;
+        if(!this.char_display.is_open) this.char_display.open(this.selected_char_index);
+        if(!this.eq_compare.is_open) this.eq_compare.open(char_key, this.selected_item.key_name);
+
+        this.control_manager.set_control(true, true, true, false, {right: this.char_display.next_char.bind(this.char_display),
+            left: this.char_display.previous_char.bind(this.char_display),
+            up: this.char_display.previous_line.bind(this.char_display),
+            down: this.char_display.next_line.bind(this.char_display),
+            esc: this.on_cancel_char_select.bind(this),
+            enter: this.on_buy_equip_select.bind(this)});
+    }
+
+    open_inventory_view(game_ticket=false){
+        if(!game_ticket && this.buy_select.is_open) this.buy_select_pos = {page: this.buy_select.current_page, index: this.buy_select.selected_index};
```

Comment:
Avoid very big lines, in the following way is much more readable:
```javascript
if(!game_ticket && this.buy_select.is_open){
  this.buy_select_pos = {
    page: this.buy_select.current_page,
    index: this.buy_select.selected_index
  };
}
```

Always prefer readability over a very compacted code.